docs/deploy.md

```diff
@@ -97,7 +97,7 @@ will need to deploy it on Kubernetes.
 
 Prerequisites:
 
-- A Kubernetes cluster running Kubernetes >=1.21
+- A Kubernetes cluster running Kubernetes >=1.23
```

Comment:
Shouldn't this be 1.24?

Reply:
i was thinking about that when i made this change... tbh, i'm not 100% sure, but my reasoning on it was:
* 1.21 was clearly an outdated comment, and it should already at least have been stating 1.22
* we just transitioned from 1.23 to 1.24.  1.23 gets end-of-lifed in october of this year. [1]
* while we aren't directly testing against 1.23 anymore, the odds are high that we'll remain compatible with it for the most part.  if we explicitly find out that we aren't, we can update the docs then.
* we never want the dev cluster to be on the oldest or the newest version of k8s.  ideally we would always be on the _most common_ version of k8s that is deployed out there at any given time.
* since it is at least possible in theory that some folks might require a slightly older version, it can't hurt us that much to support one version prior to the one we use most commonly.
* so we stay on 1.24, but we support 1.23 or later.

happy to hear input on this regardless.

[1] https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-release-calendar

Reply:
Please make it explicit that we officially support 1.24, but that we should still be backwards-compatible with 1.23, so that users won't have an unpleasant surprise.

Reply:
Please make it explicit that we officially support 1.24, but that we should still be backwards-compatible with 1.23, so that users won't have an unpleasant surprise.

Reply:
done. updated to say:
`- A Kubernetes cluster running Kubernetes >=1.23.  1.24 recommended as officially supported.`